packages/autocomplete-plugin-query-suggestions/src/getTemplates.ts

```diff
@@ -11,33 +10,75 @@ export function getTemplates<TItem extends QuerySuggestionsHit>({
   onTapAhead,
 }: GetTemplatesParams<TItem>): SourceTemplates<TItem> {
   return {
-    item({ item, root }) {
-      const content = document.createElement('div');
-      content.className = 'aa-ItemContent';
-      const icon = document.createElement('div');
-      icon.className = 'aa-ItemSourceIcon';
-      icon.innerHTML = searchIcon;
-      const title = document.createElement('div');
-      title.className = 'aa-ItemTitle';
-      title.innerHTML = reverseHighlightHit<QuerySuggestionsHit>({
-        hit: item,
-        attribute: 'query',
+    item({ item, pragma, pragmaFrag }) {
+      return pragma(pragmaFrag, {
+        children: [
+          pragma('div', {
+            className: 'aa-ItemContent',
+            children: [
+              pragma('div', {
+                className: 'aa-ItemSourceIcon',
+                children: [
+                  pragma(
+                    'svg',
+                    {
+                      width: '20',
+                      height: '20',
+                      viewBox: '0 0 20 20',
+                    },
+                    pragma('path', {
+                      d:
+                        'M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z',
+                      stroke: 'currentColor',
+                      fill: 'none',
+                      'fill-rule': 'evenodd',
+                      'stroke-width': '1.4',
+                      'stroke-linecap': 'round',
+                      'stroke-linejoin': 'round',
+                    })
+                  ),
+                ],
+              }),
+              pragma('div', {
+                className: 'aa-ItemTitle',
+                dangerouslySetInnerHTML: {
+                  __html: reverseHighlightHit({
```

Comment:
now that we're using jsx, we should use the non-joined version of the highlight and map over it, so no dangerous html is needed anymore. Can be done in a subsequent PR

Reply:
See 3a9d69b39f51dc021980a764e315f2af0195611e.